www/src/pages/editor/editDedupSite/EditDedupMineralSite.tsx

```diff
@@ -254,15 +256,66 @@ export const EditDedupMineralSite = observer(({ dedupSite, commodity }: EditDedu
         title: "Source",
         key: "reference",
         render: (_: any, site: MineralSite) => {
-          return <ReferenceComponent site={site} />;
+          const sourceId = site.sourceId;
+          let connection = sourceConnections[sourceId];
+          if (connection != null && connection != undefined && connection.startsWith("pdf:::")) {
+            connection = connection.replace("pdf:::", "");
+            const recordId = site.recordId;
+            const pageInfo = site.reference[0].pageInfo;
+            if (pageInfo != undefined || pageInfo != null) {
+              const page = pageInfo[0].page
+              connection = connection
+                .replace("{record_id}", recordId)
+                .replace("{page_number=1}", page.toString())
+            }
+            else {
+              connection = connection
+                .replace("{record_id}", recordId)
+            }
+          }
+
+          return connection ? (
+            <Typography.Link target="_blank" href={connection}>
+              {connection}
+            </Typography.Link>
+          ) : (
+            <ReferenceComponent site={site} />
+          );
```

Comment:
Move this to a separate component as the logic is complicated.

also your code need to handle interpolation correctly -- what if users enter `{page_number=2}` or `{page_number}` --- or if there is no page number.